packages/SwingSet/src/vats/comms/clist-inbound.js

```diff
@@ -27,6 +27,44 @@ export function makeInbound(state, stateKit) {
 
   // *-LocalForRemote: receiving an object/promise from a remote machine
 
+  function retireRemotePromiseID(remoteID, rpid) {
+    insistRemoteType('promise', rpid);
+    const remote = getRemote(state, remoteID);
+    const lpid = remote.fromRemote.get(rpid);
+    assert(lpid, X`unknown remote ${remoteID} promise ${rpid}`);
+    const p = state.promiseTable.get(lpid);
+    assert(
+      !p.subscribers || p.subscribers.indexOf(remoteID) === -1,
+      X`attempt to retire remote ${remoteID} subscribed promise ${rpid}`,
+    );
+    remote.fromRemote.delete(rpid);
+    remote.toRemote.delete(lpid);
```

Comment:
This `toRemote` side should be deleted already, in `beginRemotePromiseIDRetirement`, right? Or is `retireRemotePromiseID` called from multiple places?

Reply:
It's called from two places, one of which is the retirement caused by the inbound resolve that doesn't involve the ack roundtrip and hence hasn't gone through `beginRemotePromiseIDRetirement`.